fastlane_bot/helpers/carbon_trade_splitter.py

```diff
@@ -0,0 +1,195 @@
+            if not self._is_carbon_trade(trade):
+                new_trade_list.append(trade)
+                continue
+
+            carbon_exchanges = self._process_carbon_trades(trade)
+            self._create_new_trades_from_carbon_exchanges(
+                carbon_exchanges, trade, new_trade_list
+            )
+
+        return new_trade_list
+
+    def _is_carbon_trade(self, trade: TradeInstruction) -> bool:
+        """Checks if a trade is on a Carbon deployment.
+
+        Args:
+            trade: a single TradeInstruction object.
+
+        Returns:
+            True if the trade is on a Carbon deployment. False otherwise.
+
+        """
+        return trade.exchange_name in self.ConfigObj.CARBON_V1_FORKS
+
+    def _get_real_tkn(self, token_address: str, token_type: str):
+        """Returns the correct token address for the trade.
+
+        This function returns the real token address for the pool. If the token isn't the native/wrapped gas token address, it just returns the token.
+        If the token is native/wrapped, it will use the token_type to return the correct address.
+
+        Args:
+            token_address: the token address
+            token_type: the self.NATIVE, self.WRAPPED, or SELF.NEITHER
+
+        Returns:
+            The correct token address for the pool.
+
+        """
+        if token_address not in [
+            self.ConfigObj.NATIVE_GAS_TOKEN_ADDRESS,
+            self.ConfigObj.WRAPPED_GAS_TOKEN_ADDRESS,
+        ]:
+            return token_address
+        return (
+            self.ConfigObj.NATIVE_GAS_TOKEN_ADDRESS
+            if token_type == self.NATIVE
+            else self.ConfigObj.WRAPPED_GAS_TOKEN_ADDRESS
+        )
+
+    def _process_carbon_trades(self, trade: TradeInstruction) -> Dict:
+        """Processes
+
+        Process Carbon trades and organize them by exchange and token type.
+
+        Args:
+            trade: a single TradeInstruction object
+
+        Returns:
+            A dictionary containing a list of
+
+        """
+        carbon_exchanges = {}
+
+        raw_tx_str = trade.raw_txs.replace("'", '"')
+
+        raw_txs = json.loads(raw_tx_str)
+
+        for _tx in raw_txs:
+            curve = trade.db.get_pool(cid=str(_tx["cid"]).split("-")[0])
+            exchange = curve.exchange_name
+            token_type = self._get_token_type(curve)
+
+            if exchange not in carbon_exchanges:
+                carbon_exchanges[exchange] = self._initialize_exchange_data()
+
+            self._update_exchange_data(
+                carbon_exchanges[exchange], token_type, _tx, trade
+            )
+
+        return carbon_exchanges
+
+    def _get_token_type(self, curve) -> str:
+        """Determines if the trade uses native or wrapped gas tokens.
+
+        Args:
+            curve: the Pool object representing the curve.
+
+        Returns:
+            A string indicating if the curve contains native gas tokens, wrapped gas tokens, or neither.
```

Comment:
What if it contains both?

Reply:
I'm not aware of any pools that contain both native + wrapped gas token. This would definitely not handle them - but I don't think we need to expect them.